### DISCUSSION TITLE
Difference between Res<Input<MouseButton>> and EventReader<MouseButtonInput>?

### DISCUSSION THREAD

**Discussion by Metadorius** [bevyengine/bevy#5491] [Q&A] [answered]
Those two seems pretty similar, what's the difference between them and which should be used when?

**Comment by kgv** [comment, marked as answer]
https://bevy-cheatbook.github.io/input.html

> For most input types (where it makes sense), Bevy provides two ways of dealing with them:
> - by checking the current state via [resources](https://docs.rs/bevy/0.7.0/bevy/ecs/system/struct.Res.html) ([input resources](https://bevy-cheatbook.github.io/builtins.html#input-handling-resources)),
> - or via [events](https://bevy-cheatbook.github.io/programming/events.html) ([input events](https://bevy-cheatbook.github.io/builtins.html#input-events)).

> [Events](https://bevy-cheatbook.github.io/programming/events.html) ([input events](https://bevy-cheatbook.github.io/builtins.html#input-events)) are a lower-level, more all-encompassing approach. Use them if you want to get all activity from that class of input device, rather than only checking for specific inputs.
